### PR TITLE
refactor: move util/fs under util/gitfs

### PR DESCRIPTION
**Pull request by rarkins**
Part of #6617

### CODE REVIEW

**Comment by rarkins** [comment]
@zharinov I can't work out why the `readLocalFile` mocked function is being passed `undefined` filenames. Can you take a look? Note: the npm manager was the only code where I needed to add any extra mocking after doing this moving/renaming.

**Comment by rarkins** [comment]
Before adding the `jest.mock` I got this:

image

File: lib/manager/npm/post-update/node-version.spec.ts [hunk 1]
```diff
@@ -1,45 +1,47 @@
 import { mocked } from '../../../../test/util';
-import * as fs_ from '../../../util/fs';
+import * as _gitfs from '../../../util/gitfs';
 import { getNodeConstraint } from './node-version';
 
-const fs = mocked(fs_);
+jest.mock('../../../util/gitfs');
+
+const gitfs = mocked(_gitfs);
```
Comment (viceice) [inline lib/manager/npm/post-update/node-version.spec.ts:7]: Should be imported from `test/util`

**Review by viceice — COMMENTED** [review]
It's because the way typescript handle imports. If you do `import * as xxx from 'xxx'` `xxx` will only have readonly properties.

**Comment by viceice** [comment]
It's a Babel issue too. For tests we use Babel to transpile ts to js. So maybe Babel makes the exports object readonly properties.

**Comment by rarkins** [comment]
Found some useful explanations here: https://stackoverflow.com/questions/53162001/typeerror-during-jests-spyon-cannot-set-property-getrequest-of-object-which

**Review by viceice — COMMENTED** [review]
Just some test changes, we should import gitfs from test/util

**Pull request merged**